docs/examples/contrib/sqlalchemy/sqlalchemy_repository_extension.py

```diff
@@ -180,6 +181,7 @@ async def create_blog(
 
 app = Litestar(
     route_handlers=[create_blog, get_blogs, get_blog_details],
+    dependencies={"blog_post_repo": Provide(provide_blog_post_repo, sync_to_thread=True)},
```

Comment:
```suggestion
    dependencies={"blog_post_repo": Provide(provide_blog_post_repo, sync_to_thread=False)},
```
I don't think we have any need to call this dependency in a separate thread.

The only time this would be necessary is if there was some synchronous I/O occurring in the `BlogPostRepository.__init__()` method. I don't think there is, so this will be much more efficient to be called as a normal sync function.

@cofin can you confirm if I'm missing something here? I also notice we do this in the `sqlalchemy_sync_repository.py` tutorial example.

Reply:
I agree that this should be updated.  We should not be doing IO in the `__init__` method now (or in the future).

Reply:
thanks for confirming - I'll open an issue for the other module :+1: